### PR TITLE
Add RetinaFace wrapper to models 

### PR DESCRIPTION
**Pull request by ivanvikhrev**

### CODE REVIEW

File: demos/object_detection_demo/main.cpp [hunk 1]
```diff
@@ -201,6 +209,9 @@ int main(int argc, char *argv[]) {
         else if (FLAGS_at == "yolo") {
             model.reset(new ModelYolo3(FLAGS_m, (float)FLAGS_t, FLAGS_auto_resize, FLAGS_yolo_af, (float)FLAGS_iou_t, labels));
         }
+        else if (FLAGS_at == "retina") {
```
Comment (eaidova) [inline demos/object_detection_demo/main.cpp]: retinaface better, the output of retinaNet has another format and can not be parsed by retinaface parser if I right remember

File: demos/common/models/src/detection_model_retinaface.cpp [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+    LockedMemory<const void> outputMapped = rawData->rmap();
+    const float *memPtr = outputMapped.as<float*>();
+
+    for (auto i : indices) {
+        auto offset = (i % anchorNum) * sz[2] * sz[3] + i / anchorNum;
+        masks->push_back(memPtr[start + offset]);
+    }
+}
+
+std::vector<int> nms(const std::vector<ModelRetinaFace::Anchor>& boxes, const std::vector<double>& scores, double thresh) {
```
Comment (eaidova) [inline demos/common/models/src/detection_model_retinaface.cpp]: do you not have already implemented NMS code for detections, I believe it is a common postprocessing step (e.g. as I know also at least used for yolo)?

File: demos/object_detection_demo/main.cpp (hunk 1 above)
Comment (ivanvikhrev) [reply demos/object_detection_demo/main.cpp]: I named it the same way as @eizamaliev did in [python object_detectin_demo](https://github.com/openvinotoolkit/open_model_zoo/blob/97c02ad141ae6fd76320372a7b1074aa31629f79/demos/python_demos/object_detection_demo/object_detection_demo.py#L47).  I agree, we should be more specific in naming.

File: demos/common/models/src/detection_model_retinaface.cpp (hunk 2 above)
Comment (ivanvikhrev) [reply demos/common/models/src/detection_model_retinaface.cpp]: No, we don't have common NMS function in this demo:(

**Pull request closed without merging**